### ISSUE TITLE
Refactor(eos_designs): Allow vtep: true without requirement for vtep_loopback_range when using loopback0 as VTEP interface

### ISSUE THREAD

**Issue by gmuloc** [aristanetworks/ansible-avd#3427]
This issue should be used to address the title only:

* Allow vtep: true without requirement for vtep_loopback_range when using loopback0 as VTEP interface
* it should raise if vtep_loopback is set to loopback 0 and MLAG is enabled (make sure to implement a negative unit test for this specific case). Add an entry in the porting guide for now.

These two following points should be addressed as follow:

1. Implement a new node_type_key setting `default_vtep_interface` and when it is loopback0, do not required `vtep_loopback_range` => new issue with low priority

2. ~Once this is done, enable `vtep: True` on WAN node_types + refactor the if statement~. #4865 changed the way various variables are being used on WAN routers to control the LAN side of things so this is not needed anymore.

**Comment by gmuloc** [comment]
Note that today we can do:

```
# L3 Leaf switch group
l3leaf:
  defaults:
    platform: 7050SX3
    vtep_loopback: Loopback0
```

but AVD still requires `vtep_loopback_range`

**Comment by gmuloc** [comment]
This originally comes from this comment https://github.com/aristanetworks/avd/pull/3388#discussion_r1425024320 

The changes we made in this PR https://github.com/aristanetworks/avd/pull/4865 may impact the point number 2
